lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2078,6 +2079,41 @@ SubExprInfo ExprEmitter::visitSV(ArrayIndexInOutOp op) {
   return {Selection, arrayPrec.signedness};
 }
 
+SubExprInfo ExprEmitter::visitTypeOp(ArrayInjectOp op) {
+  unsigned n = op.getType().cast<ArrayType>().getSize();
```

Comment:
There's no way you'd know this, because I owe the [documentation](https://circt.llvm.org/docs/RationaleHW/#type-declarations), but please use `hw::type_cast` instead of `<some hw type>.cast`. The bottom line is this lets it work with type aliased types (i.e. human names that will emit an SV `typedef`). The longer story is we are doing [this](https://clang.llvm.org/docs/InternalsManual.html#canonical-types), and while we have deviated slightly, this part applies: "The only hard part here is remembering not to use the isa/cast/dyn_cast operations."

Reply:
Thanks for pointing this out!